### PR TITLE
Ensure works in Node.js env, allow props to be set externally, provide options for perspective, fix bug with dampening

### PR DESCRIPTION
**Pull request by willbamford**
Appreciate there are a lot of changes here. Please let me know if you want to merge only part of this and I can break it down.

### CODE REVIEW

File: regl-camera.js [hunk 1]
```diff
@@ -31,26 +36,29 @@ function createCamera (regl, props_) {
...
+
+    mouseWheel(function (dx, dy) {
+      ddistance += dy / window.innerHeight
+    })
+  }
 
   function damp (x) {
     var xd = x * 0.9
-    if (xd < 0.1) {
+    if (Math.abs(xd) < 0.1) {
```
Comment (willbamford) [inline regl-camera.js:62]: Without this swipe right with mouse has momentum but left would stop dead in it's tracks.

File: regl-camera.js [hunk 2]
```diff
@@ -16,7 +18,13 @@ function createCamera (regl, props_) {
     phi: props.phi || 0,
     distance: Math.log(props.distance || 10.0),
     eye: new Float32Array(3),
-    up: new Float32Array(props.up || [0, 1, 0])
+    up: new Float32Array(props.up || [0, 1, 0]),
+    fovy: props.fovy || Math.PI / 4.0,
+    near: typeof props.near !== 'undefined' ? props.near : 0.01,
+    far: typeof props.far !== 'undefined' ? props.far : 1000.0,
+    flipY: !!props.flipY,
```
Comment (willbamford) [inline regl-camera.js:25]: When I render out in Node.js environment and encode to PNG/JPEG etc. the image is upside down. This is a quick means of flipping the projection vertically.

File: regl-camera.js [hunk 3]
```diff
@@ -60,10 +69,16 @@ function createCamera (regl, props_) {
     return Math.min(Math.max(x, lo), hi)
   }
 
-  function updateCamera () {
+  function updateCamera (props) {
+    Object.keys(props).forEach(function (prop) {
+      cameraState[prop] = props[prop]
```
Comment (willbamford) [inline regl-camera.js:74]: Not sure of the desirability of this but I really want to be able to drive camera movement without using mouse interaction e.g. `camera({ dtheta: 0.1 }, () => {})`.

**Comment by mikolalysenko** [comment]
This looks pretty good!  Thanks!

**Pull request merged**